plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/types/ClassEmbeddingDetails.kt

```diff
@@ -3,40 +3,41 @@
  * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
  */
 
-package org.jetbrains.kotlin.formver.embeddings
+package org.jetbrains.kotlin.formver.embeddings.types
 
+import org.jetbrains.kotlin.formver.embeddings.FieldEmbedding
 import org.jetbrains.kotlin.formver.names.*
 import org.jetbrains.kotlin.formver.viper.ast.PermExp
 import org.jetbrains.kotlin.formver.viper.ast.Predicate
 import org.jetbrains.kotlin.utils.addToStdlib.ifTrue
 
 class ClassEmbeddingDetails(val type: ClassTypeEmbedding, val isInterface: Boolean) : TypeInvariantHolder {
-    private var _superTypes: List<TypeEmbedding>? = null
-    val superTypes: List<TypeEmbedding>
-        get() = _superTypes ?: error("Super types of ${type.className} have not been initialised yet.")
+    private var _superTypes: List<PretypeEmbedding>? = null
+    val superTypes: List<PretypeEmbedding>
+        get() = _superTypes ?: error("Super types of ${type.name} have not been initialised yet.")
 
     private val classSuperTypes: List<ClassTypeEmbedding>
-        get() = superTypes.filterIsInstance<ClassTypeEmbedding>()
+        get() = superTypes.mapNotNull { it as? ClassTypeEmbedding }
```

Comment:
Is there any difference (just wondering)?

Reply:
Huh, good question, I don't remember how this change happened.  I think maybe it was `it.pretype as? ClassTypeEmbedding` at some point and then I never changed it back.  Thanks for spotting it!